### PR TITLE
Transactional outbox

### PR DESCRIPTION
**Pull request by jjmschofield**

### CODE REVIEW

File: patterns/transactional-outbox/src/producer/lib/messages/fail.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {MessageRecord} from "./types";
+import {MessageModel} from "./model";
+
+export const recordFailure = async (message: MessageRecord) => {
+  const record = await MessageModel.findOne({
+    where: {id: message.id}
+  });
+
+  if(!record) throw new Error('pending message not found');
```
Comment (garyposter) [inline patterns/transactional-outbox/src/producer/lib/messages/fail.ts:9]: This would not be desirable in production, yeah? Or am I missing something?

Comment (garyposter) [reply patterns/transactional-outbox/src/producer/lib/messages/fail.ts:9]: Maybe Boom handles it?

File: patterns/transactional-outbox/src/producer/lib/messages/get.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      lastAttemptAt: {
+        [Op.or]: {
+          [Op.eq]: null,
+          [Op.lt]: new Date(Date.now() - retryDelayMs)
+        },
+      },
+      lockedAt: {
+        [Op.or]: {
+          [Op.eq]: null,
+          [Op.lt]: new Date(Date.now() - retryDelayMs)
```
Comment (garyposter) [inline patterns/transactional-outbox/src/producer/lib/messages/get.ts:41]: This is all nice. :-)

File: patterns/transactional-outbox/src/producer/lib/messages/fail.ts (hunk 1 above)
Comment (jjmschofield) [reply patterns/transactional-outbox/src/producer/lib/messages/fail.ts]: yeah boom normalizes the error into a standard shape for all responses, can't remember if it gives out revealing info

**Pull request merged**